src/server/worker/master.go

```diff
@@ -692,6 +692,7 @@ func (a *APIServer) waitJob(pachClient *client.APIClient, jobInfo *pps.JobInfo,
 			}); err != nil && !pfsserver.IsCommitFinishedErr(err) {
 				return err
 			}
+			return nil
```

Comment:
I could be wrong but I think we still want to do some of the stuff below this when there's a failed job. Such as writing out the datums object. @brycemcanally can clarify.

Reply:
We should be returning here, this is correct. The path @msteffen described is intended for killing a job not a failed datum, it just so happens that it would cancel the ctx (because the commit would be finished) and would work out correctly some (I guess most?) of the time. I introduced this issue when making the short term job logic changes because I added the logic for not considering the finished commit error and accidentally did not keep a return here.